TestIngestValidation fails under stress

This test fails right away under stress but not if I just run it normally repeatedly. I believe the test creates a directory inside the local tree and can't be run in parallel like `stress` does. We should make sure it uses a temp dir with a unique name. CC @nicktrav 

```
radu (master)+ /go/src/github.com/cockroachdb/pebble # go test . -run TestIngestValidation -count 1 -exec 'stress'


--- FAIL: TestIngestValidation (0.03s)
    ingest_test.go:2226: rng seed = 1686874341769169600
    --- FAIL: TestIngestValidation/start_block (0.01s)
        ingest_test.go:2302: 
                Error Trace:    ingest_test.go:2302
                                                        ingest_test.go:2340
                                                        ingest_test.go:2387
                Error:          Received unexpected error:
                                open ext: no such file or directory
                Test:           TestIngestValidation/start_block
    --- FAIL: TestIngestValidation/non-end_block (0.01s)
        ingest_test.go:2282: 
                Error Trace:    ingest_test.go:2282
                                                        ingest_test.go:2340
                                                        ingest_test.go:2387
                Error:          Received unexpected error:
                                pebble/table: invalid table (bad magic number: 0x0f9a68868d7eeb29)
                                (1) forced error mark
                                  | "pebble: corruption"
                                  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
                                Wraps: (2) attached stack trace
                                  -- stack trace:
                                  | github.com/cockroachdb/pebble/internal/base.CorruptionErrorf
                                  |     /go/src/github.com/cockroachdb/pebble/internal/base/error.go:27
                                  | github.com/cockroachdb/pebble/sstable.readFooter
                                  |     /go/src/github.com/cockroachdb/pebble/sstable/table.go:384
                                  | github.com/cockroachdb/pebble/sstable.NewReader
                                  |     /go/src/github.com/cockroachdb/pebble/sstable/reader.go:3867
                                  | github.com/cockroachdb/pebble.TestIngestValidation.func1.3
                                  |     /go/src/github.com/cockroachdb/pebble/ingest_test.go:2281
                                  | github.com/cockroachdb/pebble.TestIngestValidation.func1.4
                                  |     /go/src/github.com/cockroachdb/pebble/ingest_test.go:2340
                                  | github.com/cockroachdb/pebble.TestIngestValidation.func1
                                  |     /go/src/github.com/cockroachdb/pebble/ingest_test.go:2387
                                  | testing.tRunner
                                  |     /usr/local/go1.19.4/src/testing/testing.go:1446
                                  | runtime.goexit
                                  |     /usr/local/go1.19.4/src/runtime/asm_amd64.s:1594
                                Wraps: (3) pebble/table: invalid table (bad magic number: 0x0f9a68868d7eeb29)
                                Error types: (1) *markers.withMark (2) *withstack.withStack (3) *errutil.leafError
                Test:           TestIngestValidation/non-end_block
FAIL
```

--- a/ingest_test.go
+++ b/ingest_test.go
@@ -11,6 +11,7 @@ import (
 	"io"
 	"math"
 	"os"
+	"path/filepath"
 	"sort"
 	"strconv"
 	"strings"
@@ -2220,6 +2221,7 @@ func TestIngestValidation(t *testing.T) {
 
 		ingestTableName = "ext"
 	)
+	ingestPath := filepath.Join(t.TempDir(), ingestTableName)
 
 	seed := uint64(time.Now().UnixNano())
 	rng := rand.New(rand.NewSource(seed))
@@ -2298,7 +2300,7 @@ func TestIngestValidation(t *testing.T) {
 				}
 				bh := l.Data[blockIdx]
 
-				osF, err := os.OpenFile(ingestTableName, os.O_RDWR, 0600)
+				osF, err := os.OpenFile(ingestPath, os.O_RDWR, 0600)
 				require.NoError(t, err)
 				defer func() { require.NoError(t, osF.Close()) }()
 
@@ -2320,9 +2322,9 @@ func TestIngestValidation(t *testing.T) {
 				// Create a disk-backed file outside of the DB FS that we can
 				// open as a regular os.File, if required.
 				tmpFS := vfs.Default
-				f, err := tmpFS.Create(ingestTableName)
+				f, err := tmpFS.Create(ingestPath)
 				require.NoError(t, err)
-				defer func() { _ = tmpFS.Remove(ingestTableName) }()
+				defer func() { _ = tmpFS.Remove(ingestPath) }()
 
 				w := sstable.NewWriter(objstorageprovider.NewFileWritable(f), sstable.WriterOptions{
 					BlockSize:   blockSize,     // Create many smaller blocks.
@@ -2335,13 +2337,13 @@ func TestIngestValidation(t *testing.T) {
 
 				// Possibly corrupt the file.
 				if tc.cLoc != corruptionLocationNone {
-					f, err = tmpFS.Open(ingestTableName)
+					f, err = tmpFS.Open(ingestPath)
 					require.NoError(t, err)
 					corrupt(f)
 				}
 
 				// Copy the file into the DB's FS.
-				_, err = vfs.Clone(tmpFS, fs, ingestTableName, ingestTableName)
+				_, err = vfs.Clone(tmpFS, fs, ingestPath, ingestTableName)
 				require.NoError(t, err)
 
 				// Ingest the external table.

